src/rime/algo/syllabifier.cc

```diff
@@ -74,20 +98,30 @@ int Syllabifier::BuildSyllableGraph(const string &input,
             props.end_pos = end_pos;
             // add a syllable with properties to the edge's
             // spelling-to-syllable map
-            spellings.insert({syllable_id, props});
+            if (matches_set.find(m.value) == matches_set.end()) {
+              props.is_correction = true;
+              props.credibility = 0.01;
```

Comment:
should it be reassigned or `*= PENALTY`, if there was an assigned credibility value for the original spelling type?